webextension/background.js

```diff
@@ -38,37 +38,41 @@ class StudyLifeCycleHandler {
    * call `this.enableFeature` to actually do the feature/experience/ui.
    */
   constructor() {
+    // IMPORTANT:  Listen for onEndStudy first.
     browser.study.onEndStudy.addListener(this.handleStudyEnding);
```

Comment:
@gregglind Tested after pulling latest shieldutils v5, Disable and uninstall of add-on doesn't trigger handleStudyEnding. Is there more to it?